apps/backend/src/modules/contributors/contributor.service.ts

```diff
@@ -45,19 +52,53 @@ class ContributorService implements Service {
 			});
 		}
 
-		return item.toObject();
+		const projects = await this.projectService.findProjectsByContributorId(id);
+
+		return { ...item.toObject(), projects: projects.items };
 	}
 
-	public findAll(): ReturnType<Service["findAll"]> {
-		return Promise.resolve({ items: [] });
+	public async findAll(): Promise<ContributorGetAllResponseDto> {
+		const contributors = await this.contributorRepository.findAll();
+
+		const contributorsWithProjects = await Promise.all(
+			contributors.items.map(async (item) => {
+				const contributor = item.toObject();
+
+				const projects = await this.projectService.findProjectsByContributorId(
```

Comment:
Maybe we can join projects inside contributors repository findAll method?

Reply:
Oh, I thought that such logic should be through services. You mean that I should remove it completely from projectRepository, and create such method in contributorRepository, or make contributorRepository dependent of projectRepository?

Reply:
No, I was thinking about calculating it as a custom field somehow inside the query in contributors repository, but looks like it will be complicated, so let's leave it as it is

Reply:
Then I think we don't need the projects field inside contributors entity, looks like we don't use it

Reply:
so the problem is that it's not really good to make a separate query for each item. Also, we need to use for of, instead of Promise all, because this may make too many simultaneous db requests, which is bad. And the concern here is that with for of it will work slow

Reply:
and if there is a possibility to compute or somehow join this inside the contributors query, this would be better, otherwise we need to use for of here, which can be slow

Reply:
Oh, alright, I'll try to figure out what we could do with it tomorrow. Resolving a lot of conflicts today got me feeling fried today😂